source/iterator/iterator.go

```diff
@@ -225,6 +230,48 @@ func (c *CombinedIterator) SetupCDC(ctx context.Context, db *sqlx.DB) error {
 	return nil
 }
 
+func (c *CombinedIterator) getCreateTrackingTableQuery(ctx context.Context, db *sqlx.DB) (string, error) {
+	rows, err := db.QueryContext(ctx, queryGetColumnsInfo, c.table)
+	if rows.Err() != nil {
+		return "", fmt.Errorf("query column information: %w", rows.Err())
+	}
+	if err != nil {
+		return "", fmt.Errorf("query column information: %w", err)
+	}
+	defer rows.Close()
+
+	// Construct the CREATE TABLE query for the new table
+	createQuery := fmt.Sprintf("CREATE TABLE %s (", c.trackingTable)
+	for rows.Next() {
+		var columnName, dataType string
+		var charMaxLength, numericPrecision, numericScale sql.NullInt64
+		if err := rows.Scan(&columnName, &dataType, &charMaxLength, &numericPrecision, &numericScale); err != nil {
+			return "", err
+		}
+		var columnDefinition string
+		switch dataType {
```

Comment:
Is the data type always going to be lowercase? If not, I propose adding `strings.ToLower(dataType)` here.